### PR TITLE
Add configuration for enableStripMarginOnTypeFormatting

### PR DESCRIPTION
**Pull request by mlachkar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -143,6 +143,11 @@
           "type": "boolean",
           "default": false,
           "description": "Enable/disable goto super method code lens."
+        },
+        "metals.enableStripMarginOnTypeFormatting": {
+          "type": "boolean",
+          "default": true,
+          "description": "When enabled, if you press the return key from the first line of a multiline string containing a pipe, it will automatically add .stripMargin."
```
Comment (ckipp01) [inline package.json:150]: Super nitpicky, but instead of using `description` here, you could use `markdownDescription` and then actually put `stripMargin` in backticks which might appear a bit nicer in vscode.

Comment (mlachkar) [reply package.json:150]: ok ! thanks for your review

File: package.json [hunk 2]
```diff
@@ -377,7 +382,7 @@
     "@types/node": "14.0.5",
     "@types/shell-quote": "1.6.2",
     "@types/vscode": "1.43.0",
-    "prettier": "2.0.5",
+    "prettier": "^2.0.5",
```
Comment (gabro) [inline package.json:385]: minor: why this change?
For context: we rely on dependabot to send patches for the updates so we prefer to use fixed versions in the `package.json`

Comment (mlachkar) [reply package.json]: I thought it was added by `yarn format`. 
I fixed it now.

Comment (gabro) [reply package.json]: thanks :)
